src/commands/reminders/remind.ts

```diff
@@ -11,38 +11,53 @@ export default class extends Command {
 	description = `Have ${BOT.NAME} give you a reminder.`;
 	usage = '<reminder> | <duration> | [repeat]';
 	extendedHelp = 'Reminders can be set to repeat daily or weekly.';
-
-	run(msg: Message, [reminder]: [Reminder]): Promise<Message> {
-		msg.client.mongo.collection(DB.REMINDERS).insertOne(reminder);
-
-		return msg.channel.send(`I'll remind you about that at ${reminderTime(reminder)}.`);
-	}
-
-	argParser(msg: Message, input: string): [Reminder] {
-		const [content, rawDuration, rawRepeat] = input.split('|').map(part => part.trim());
-		const weekWords = ['w', 'week', 'weekly'];
-		const dayWords = ['d', 'day', 'daily'];
-
+	options: ApplicationCommandOptionData[] = [
+		{
+			name: 'content',
+			description: 'What you\'d like to be reminded of',
+			type: 'STRING',
+			required: true
+		},
+		{
+			name: 'duration',
+			description: 'When you\'d like to be reminded',
+			type: 'STRING',
+			required: true
+		},
+		{
+			name: 'repeat',
+			description: 'How often you want the reminder to repeat',
+			choices: [{ name: 'daily', value: 'daily' }, { name: 'weekly', value: 'weekly' }],
+			type: 'STRING',
+			required: false
+		}
+	]
+
+	tempRun(interaction: CommandInteraction): Promise<unknown> {
+		const content = interaction.options.getString('content');
+		const rawDuration = interaction.options.getString('duration');
 		const duration = parse(rawDuration);
-		if (!duration) throw `**${rawDuration}** is not a valid duration.`;
-
-		const repeat = rawRepeat
-			? weekWords.includes(rawRepeat.toLowerCase())
-				? 'weekly'
-				: dayWords.includes(rawRepeat.toLowerCase())
-					? 'daily'
-					: 'error'
-			: null;
-
-		if (repeat === 'error') throw `**${rawRepeat}** is not a valid repeat value.`;
-
-		return [{
-			owner: msg.author.id,
+		const repeat = interaction.options.getString('repeat') as 'daily' | 'weekly' || null;
+
+		if (!duration) {
+			return interaction.reply({
+				content: `**${rawDuration}** is not a valid duration. You can use words like hours, minutes, seconds, days, weeks, months, or years.`,
+				ephemeral: true
+			});
+		}
+		const reminder: Reminder = {
+			owner: interaction.user.id,
 			content,
-			mode: msg.channel.type === 'DM' ? 'private' : 'public',
+			mode: interaction.channel.type === 'DM' ? 'private' : 'public',
```

Comment:
Would it be worth checking if the user has DMs open at some point during the reminder setup process?

Reply:
I'm not sure what you mean by this, could you elaborate?

Reply:
> Would it be worth checking if the user has DMs open at some point during the reminder setup process?

The user can't send an interaction with the channel type as `DM` if they don't have DMs open.

Reply:
Yeah if you don't catch an error by checking, then the whole thing will throw an error if DMs are closed